scene/main/canvas_layer.cpp

```diff
@@ -87,6 +87,17 @@ Transform2D CanvasLayer::get_transform() const {
 	return transform;
 }
 
+Transform2D CanvasLayer::get_final_transform() const {
+	Transform2D follow;
+	if (is_following_viewport()) {
+		follow.scale(Vector2(get_follow_viewport_scale(), get_follow_viewport_scale()));
+		if (vp) {
+			follow = vp->get_canvas_transform() * follow;
+		}
```

Comment:
Should follow scale be applied if `vp` is invalid? :thinking:

Reply:
This is a philosophical question. I would be happy to adjust the PR, acording to the preference of the maintainers.